extension/js/common/core/crypto/smime/smime-key.ts

```diff
@@ -6,21 +6,23 @@ import { UnreportableError } from '../../../platform/catch.js';
 import { PgpArmor } from '../pgp/pgp-armor.js';
 import { Buf } from '../../buf.js';
 import { MsgBlockParser } from '../../msg-block-parser.js';
+import { MsgBlock } from '../../msg-block.js';
 
 export class SmimeKey {
 
   public static parse = (text: string): Key => {
     if (text.includes(PgpArmor.headers('certificate').begin)) {
       const blocks = MsgBlockParser.detectBlocks(text).blocks;
       const certificates = blocks.filter(b => b.type === 'certificate');
-      if (certificates.length < 1) { // todo: should we select the correct certificate if a chain is provided?
+      const leafCertificates = SmimeKey.getLeafCertificates(certificates);
+      if (leafCertificates.length !== 1) {
         throw new Error('Could not parse S/MIME key without a certificate');
       }
```

Comment:
Maybe first test for > 1 leafs to throw `Parsing S/MIME with more than one user certificate is not supported`

Reply:
Just for clarity, else it may look like there wasn't any cert when there were two